File: elixir/7ku/fizz_buzz.ex
defmodule FizzBuzz do
  def fizzbuzz(n) do
    1..n
    |> Enum.map(fn x ->
      cond do
        rem(x, 3) == 0 and rem(x, 5) == 0 -> "FizzBuzz"
        rem(x, 5) == 0 -> "Buzz"
        rem(x, 3) == 0 -> "Fizz"
        x -> x
      end
    end)
  end
end

File: ruby/7ku/Fizz Buzz.rb
# frozen_string_literal: true

# https://www.codewars.com/kata/5300901726d12b80e8000498/train/ruby

def fizzbuzz(n)
  (1..n).map do |i|
    if (i % 3).zero? && (i % 5).zero?
      'FizzBuzz'
    elsif (i % 5).zero?
      'Buzz'
    elsif (i % 3).zero?
      'Fizz'
    else
      i
    end
  end
end

p fizzbuzz(15) #== [1, 2, 'Fizz', 4, 'Buzz', 'Fizz', 7, 8, 'Fizz', 'Buzz']
